frontend/src/components/AppBreadCrumb.vue

```diff
@@ -7,35 +7,56 @@
 </template>
 
 <script>
+import { mapGetters } from 'vuex'
+
 export default {
+    computed: {
+        ...mapGetters('subject', {
+            subjects: 'subjectMap'
+        }),
+        routeList() {
+            // e.g. Subjects > Computer Science > COMP4920 > Ask Question
+
+            let id = 0
+            const list = [
+                {id: id++, text: 'Home', route: '/'},
+                {id: id++, text: 'Subjects', route: '/subject'}
+            ]
+
+            const path = this.$route.path.split('/')
+            if (path.length) {
+                if (this.$route.params.code) {
+                    // if there's a course or subject code, add the subject name
+                    const subjCode = this.$route.params.code.slice(0,4)
```

Comment:
Shouldn't this be `substr`?

Reply:
yeah I meant substring(). Yeah I guess it's just more explicit as we are talking about a string. But it's doesn't really bother me.